### DISCUSSION TITLE
How to inject Fastenv to FastAPI methods

### DISCUSSION THREAD

**Discussion by clabnet** [br3ndonland/fastenv#28] [Q&A] [answered]
Hello, 
I have a FastAPI project, and Fastenv load fine a .env file during FastAPI startup:
```
async def lifespan(app: FastAPI):
    """Loading .env file"""
    mysettings = await dot_env.load_my_dotenv()
```
Sorry for dumb question: 
- how to pass `mysettings` to other FastAPI methods ? 
(to avoid read multiple times the same .env file)

Thank you in advance

**Comment by br3ndonland** [comment, marked as answer]
Thanks for your interest @clabnet. It's a good question and I was thinking I should write some docs to more clearly explain this topic.

I agree with your suggestion of using a [lifespan event](https://fastapi.tiangolo.com/advanced/events/). I would just modify it slightly. To access your `DotEnv` model after the application starts up, maybe you could set it as an attribute on your `FastAPI` instance.

```py
from contextlib import asynccontextmanager

import fastenv
from fastapi import FastAPI


@asynccontextmanager
async def lifespan(app: FastAPI):
    """Load .env file"""
    settings = await fastenv.load_dotenv(".env", find_source=True)
    app.settings = settings
    yield


app = FastAPI(lifespan=lifespan)
```

Let me know how it goes.

**Reply by clabnet** [reply]
Yes it work, thank's

**Reply by br3ndonland** [reply]
I've added an improved example to the quickstart in the README and [docs](https://fastenv.bws.bio/#quickstart) using [lifespan state](https://www.starlette.io/lifespan/#lifespan-state) instead of setting an attribute on the `FastAPI` app instance. Either way will work, but lifespan state is the recommended way to share objects between the lifespan function and API endpoints.
